dev-docs/source/Workbench/RunningWorkbench.rst

```diff
@@ -0,0 +1,34 @@
+.. _RunningWorkbench:
+
+=================
+Running Workbench
+=================
+
+How does Workbench get run
+##########################
+
+Mantid Workbench is run on two separate Processes. The first process, known as the parent process, creates a second process using the
+`multiprocessing python package <https://docs.python.org/3/library/multiprocessing.html>`_, known as the child process. The responsibilities
+of the two processes are roughly as follows:
+
+- The child process is the process on which the Workbench is opened. If there is a terminating fault that occurs when using the Workbench,
+such as a segmentation fault, this process will exit. Alternatively, a close event triggered by a user will also cause this process to exit.
+
+- The responsibility of the parent process is to wait for the child process to exit. When the child process exits, a decision is made whether
+an ErrorReporter window needs to be opened depending on the exit code of the Workbench.
+
+A note on debugging
+###################
+
+Debuggers can sometimes attach to the parent process, causing your breakpoints not to be hit. To avoid this, we can run the Workbench using the ``--single-process`` flag which will ensure only one process is used to run the Workbench.
```

Comment:
I think this sounds a little like what happens is random. Can we say that the default behaviour of Workbench is to start as a parent and child process as described above but for debugging we want to attach to the main process users see (the child in the above scenario). To accomplish this we can disable the two-process mode using the `--single-process` flag.